Feature/unidade receptora lotacao hotfix10.0.73

michelrisucci-codata opened the pull request.

wendell-codata committed:
Inclusão de Unidade Receptora em Lotação

wendell-codata committed:
Correção consulta unidade receptora

michelrisucci-codata commented on siga-cp/src/main/java/br/gov/jfrj/siga/dp/AbstractDpLotacao.java:
```diff
@@ -184,6 +184,9 @@ public abstract class AbstractDpLotacao extends DpResponsavel implements
 	private Integer isSuspensa;
 
+	@Column(name = "UNIDADE_RECEPTORA")
+	private boolean unidadeReceptora;
```
@wendell-codata isso precisa de script SQL, não? Adiciona uma migration com isso pra ficar registrado.

wendell-codata replied:
Seque script a ser executado, o mesmo ja insere valor default FALSE para todas as colunas.

```sql
alter table corporativo.dp_lotacao add "unidade_receptora" bool default false not null;
```

michelrisucci-codata replied:
@wendell-codata preciso desse script num arquivo de migration pra a gente não deixar solto aqui.

wendell-codata replied:
Foi realizada a criação do arquivo conforme orientação acima.

michelrisucci-codata commented on siga-cp/src/main/java/br/gov/jfrj/siga/cp/bl/CpBL.java:
```diff
@@ -1769,7 +1769,7 @@ public void enviarEmailTokenResetPIN(DpPessoa destinatario, String assunto, Stri
 	public DpLotacao criarLotacao(final CpIdentidade identidadeCadastrante, final DpPessoa titular, final DpLotacao lotaTitular, 
 			final Long id, final String nmLotacao, final Long idOrgaoUsu, final String siglaLotacao,
-			final String situacao, final Boolean isExternaLotacao, final Long lotacaoPai, final Long idLocalidade) {
+			final String situacao, final Boolean isExternaLotacao, final Long lotacaoPai, final Long idLocalidade, final Boolean unidadeReceptora) {
```
Existe uma motivação para estar usando `Boolean` ao invés de `boolean` aqui? O `equals()` da linha 1823 pode dar problema se tiver nulo.

wendell-codata replied:
Substituido pra tipo primario

michelrisucci-codata commented on siga-cp/src/main/java/br/gov/jfrj/siga/dp/dao/CpDao.java:
```diff
@@ -738,9 +738,25 @@ private JPAQuery<?> queryConsultarPorFiltro(final DpLotacaoDaoFiltro filtro) {
-				predicates.and(qCpOrgaoUsuario.idOrgaoUsu.eq(filtro.getIdOrgaoUsu()));
-			}
+			 if (filtro.getIdOrgaoUsu() != null && filtro.getIdOrgaoUsu().longValue() > 0) {
+	                predicates.and(qCpOrgaoUsuario.idOrgaoUsu.eq(filtro.getIdOrgaoUsu()));
```
Indentação não tá legal nesse bloco de IF. Como é código novo, melhor manter o padrão. Dê um `Ctrl+Shift+F` no Eclipse, apenas nesta seção (selecionando entre a linha 741 e 759 (não na página toda se não vai bagunçar muito). Se tiver dúvida, me chama.

wendell-codata replied:
Identação era mostrada normalmente na IDE... mas percebi que não estava utilizando TAB... removi e tabulei.

michelrisucci-codata commented on siga-cp/src/main/java/br/gov/jfrj/siga/dp/dao/CpDao.java:
```diff
@@ -738,9 +738,25 @@ private JPAQuery<?> queryConsultarPorFiltro(final DpLotacaoDaoFiltro filtro) {
+	                        .from(subqDpLotacaoCountUndRcpt)
+	                        .where(subqDpLotacaoCountUndRcpt.orgaoUsuario.idOrgaoUsu.eq(filtro.getIdOrgaoUsu())
+	                                .and(subqDpLotacaoCountUndRcpt.unidadeReceptora.isTrue()))
+	                        .eq(0L);
```
Esta condicional
```java
	final BooleanExpression predicateOrgaoSemUnidade = JPAExpressions
                        .select(subqDpLotacaoCountUndRcpt.idLotacao.count())
                        .from(subqDpLotacaoCountUndRcpt)
                        .where(subqDpLotacaoCountUndRcpt.orgaoUsuario.idOrgaoUsu.eq(filtro.getIdOrgaoUsu())
                                .and(subqDpLotacaoCountUndRcpt.unidadeReceptora.isTrue()))
                        .eq(0L);
```
Foi aquela que chegamos à conclusão de que é a adequada pra adaptar a query?

wendell-codata replied:
Sim.... mas foi removida, pois não houve mais a necessidade.

michelrisucci-codata commented on siga/src/main/java/br/gov/jfrj/siga/vraptor/DpPessoaController.java:
```diff
@@ -250,14 +250,14 @@ public void lista(Integer paramoffset, Long idOrgaoUsu, String nome, String cpfP
 				final CpOrgaoUsuario primeiroOrgaoUsuario = Iterables.getFirst(list, null);
 				final Long primeiroIdOrgaoUsuario = primeiroOrgaoUsuario != null ? primeiroOrgaoUsuario.getId() : null;
-				carregarCombos(null, primeiroIdOrgaoUsuario, null, null, null, null, null, 0, Boolean.FALSE);
+				carregarCombos(null, primeiroIdOrgaoUsuario, null, null, null, null, null, 0, Boolean.FALSE, Boolean.FALSE);
```
Por isso perguntei. Caso esse booleano seja não nulo, melhor deixar `boolean` ao invés de `Boolean` porque amarra o parâmetro não nulo.

wendell-codata replied:
Realizei a alteração para `boolean`, seguindo as orientações acima.

wendell-codata committed:
Remoção de linhas não utilizadas e altreração do Boolean

wendell-codata committed:
Correção realizada em função da substituição do tipo Boolean

wendell-codata committed:
Correção de parâmetro que estava sendo enviado errado na API.

wendell-codata committed:
Remoção da opção [Todos] na listagem de Órgãos em edição de documentos.

wendell-codata committed:
Ajuste em buscar todos os órgãos de acordo com unidade receptora

wendell-codata committed:
Inclusão de Script SQL de inclusão de coluna unidade receptora no migration do sigadoc

wendell-codata committed:
Inclusão de script na unidade receptora nova nomenclatura.

michelrisucci-codata committed:
[FEATURE] Unidade Receptora: ajuste padrão POJO

michelrisucci-codata committed:
[FEATURE] Unidade Receptora: ajuste de padronização de código

wendell-codata committed:
Alteração de tipo de dado Boolean para boolean

wendell-codata committed:
Alteração do tipo de dado em CPOrgaoUsuario para Long

wendell-codata committed:
Alteração de tipo de dado Integer para Long corrigindo problema

michelrisucci-codata committed:
Merge branch 'hotfix/10.0.73' into feature/unidade-receptora-lotacao-hotfix10.0.73
